react/components/ProductQuantityStepper.js

```diff
@@ -1,64 +1,106 @@
+        seller
+        index
+        parentItemIndex
+        parentAssemblyBinding
+      }
+    }
+  }
+`
+
+export const UPDATE_ITEMS_MUTATION = gql`
+  mutation updateItems($items: [MinicartItem]) {
+    updateItems(items: $items) @client
+  }
+`
+
 class ProductQuantityStepper extends Component {
   static propTypes = {
     product: productShape.isRequired,
     onUpdateItemsState: PropTypes.func.isRequired,
     showToast: PropTypes.func,
     intl: intlShape,
+    minicartItems: PropTypes.array,
+    updateItems: PropTypes.func.isRequired,
   }
 
   state = {
     quantity: this.props.product.quantity,
     canIncrease: true,
   }
 
-  handleOnChange = (e) => {
+  componentDidUpdate = prevProps => {
+    const {
+      product: { quantity: prevQuantity },
+    } = prevProps
+    const {
+      product: { quantity },
+    } = this.props
+    if (prevQuantity !== quantity) {
+      this.setState({ quantity })
+    }
+  }
+
+  handleOnChange = e => {
     e.stopPropagation()
     e.preventDefault()
     this.props.onUpdateItemsState(true)
-    this.setState({ quantity: e.value }, () => this.debouncedUpdate(this.state.quantity))
+    this.setState({ quantity: e.value }, () =>
+      this.debouncedUpdate(this.state.quantity)
+    )
   }
 
   checkUpdatedQuantity = (updateResponse, itemIndex, expectedQuantity) => {
     const { showToast, intl } = this.props
     const actualQuantity = updateResponse.items[itemIndex].quantity
     if (actualQuantity !== expectedQuantity) {
       this.setState({ canIncrease: false, quantity: actualQuantity })
-      showToast({ message: intl.formatMessage({ id: 'editor.productSummary.quantity-error' }) })
+      showToast({
+        message: intl.formatMessage({
+          id: 'editor.productSummary.quantity-error',
+        }),
+      })
     }
   }
 
-  updateItemQuantity = async (quantity) => {
-    const { product, orderFormContext } = this.props
+  updateItemQuantity = async quantity => {
+    const { product, minicartItems, updateItems } = this.props
     this.setState({ canIncrease: true })
-    const itemIndex = findIndex(propEq('id', product.sku.itemId))(orderFormContext.orderForm.items)
+    const itemIndex = findIndex(propEq('id', product.sku.itemId), minicartItems)
     try {
-      await orderFormContext.updateOrderForm({
-        variables: {
-          orderFormId: orderFormContext.orderForm.orderFormId,
-          items: [{
-            index: itemIndex,
-            id: product.sku.itemId,
-            quantity,
-            seller: product.sku.sellerId,
-          }],
+      await updateItems([
+        {
+          ...minicartItems[itemIndex],
+          index: itemIndex,
+          quantity,
+          seller: product.sku.sellerId,
         },
-      })
-      const orderForm = await orderFormContext.refetch()
-      this.checkUpdatedQuantity(orderForm.data.orderForm, itemIndex, quantity)
+      ])
     } catch (err) {
       // gone wrong, rollback to old quantity value
```

Comment:
What about this?

Reply:
Resolved with [this](https://github.com/vtex-apps/minicart/pull/116/commits/11bf5df1aeda1ce6cd0d7a554ecb40e5fe4798ae)

Reply:
Deleted after the force push, but I trust you, bro.

Reply:
Oh... [There it is](https://github.com/vtex-apps/minicart/pull/116/commits/408d41ed9535e6ff088f5d9f6e9f6aff304097b7)